components/eamxx/src/diagnostics/histogram.cpp

```diff
@@ -0,0 +1,156 @@
+#include "diagnostics/histogram.hpp"
+
+#include <ekat_team_policy_utils.hpp>
+
+
+namespace scream {
+
+HistogramDiag::HistogramDiag(const ekat::Comm &comm, const ekat::ParameterList &params)
+    : AtmosphereDiagnostic(comm, params) {
+  const auto &field_name = m_params.get<std::string>("field_name");
+  m_bin_configuration    = params.get<std::string>("bin_configuration");
+  m_diag_name            = field_name + "_histogram_" + m_bin_configuration;
+}
+
+void HistogramDiag::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
+  const auto &field_name = m_params.get<std::string>("field_name");
+  const auto &grid_name  = m_params.get<std::string>("grid_name");
+  add_field<Required>(field_name, grid_name);
+}
+
+void HistogramDiag::initialize_impl(const RunType /*run_type*/) {
+  using FieldIdentifier = FieldHeader::identifier_type;
+  using FieldLayout     = FieldIdentifier::layout_type;
+  using ShortFieldTagsNames::CMP;
+  const Field &field              = get_fields_in().front();
+  const FieldIdentifier &field_id = field.get_header().get_identifier();
+  const FieldLayout &field_layout = field_id.get_layout();
+  EKAT_REQUIRE_MSG(field_layout.rank() >= 1 && field_layout.rank() <= 3,
+                   "Error! Field rank not supported by HistogramDiag.\n"
+                   " - field name: " +
+                       field_id.name() +
+                       "\n"
+                       " - field layout: " +
+                       field_layout.to_string() + "\n");
+
+  // extract bin values from configuration
+  std::istringstream stream(m_bin_configuration);
+  std::string token;
+  std::vector<Real> bin_values;
```

Comment:
FYI, you can just do
```c++
auto bin_values = ekat::split(m_bin_configuration,"_");
```
This utility is in the header `ekat_string_utils.hpp`, and mimics python's strings split method.

Reply:
I think we recently had a PR going in to fix the missing includes in the ml correction processes.